src/EntityDiagram/ExpandedDiagram.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import {hierarchy, Tree} from "@visx/hierarchy";
+import {LinearGradient} from "@visx/gradient";
+import {Group} from "@visx/group";
+import {Text} from "@visx/text";
+import OffsetLine from "./OffsetLine"
+import {StudyData} from "./Types";
+
+interface ShadingData {
+  entityId: string
+  value: number
+  color?: string
+}
+
+interface ExpandedDiagram {
+  treeData: StudyData
+  orientation: string
+  highlightedEntityID: string
+  shadingData: ShadingData[]
```

Comment:
I think this should be an object keyed by entityId, not an array. This way, lookups are faster.

```suggestion
  shadingData: Record<string, ShadingData | undefined>
```